spring-security-cas-extension/src/main/java/com/kakawait/spring/security/cas/client/AuthenticatedPrincipal.java

```diff
@@ -0,0 +1,11 @@
+package com.kakawait.spring.security.cas.client;
+
+import java.security.Principal;
+
+/**
+ * @author Jonathan Coueraud
+ */
+public interface AuthenticatedPrincipal {
+
+    Principal getAuthenticatedPrincipal();

```

Comment:
How to tell that is not possible to retrieve `Principal` (for any reasons)? Maybe think about one of this proposition (order is totally random):

1. Use `Optional` 
2. Document to tell that you could return `null` if not possible to retrieve principal
3. Add _checked exception_
4. Add documented _runtime exception_